winch/codegen/src/isa/x64/masm.rs

```diff
@@ -1043,6 +1047,7 @@ impl MacroAssembler {
     fn store_impl(&mut self, src: RegImm, dst: Address, size: OperandSize, flags: MemFlags) {
         let scratch = <Self as Masm>::ABI::scratch_reg();
         let float_scratch = <Self as Masm>::ABI::float_scratch_reg();
+        let vector_scratch = <Self as Masm>::ABI::vector_scratch_reg();
```

Comment:
I've been giving a bit of thought to this change. If I'm not wrong the motivation is to future-proof, be explicit and correctly differentiate the scratch register for different types (GPR, FPR) and sizes (FPR vs v128). 

I'm wondering if there's an alternative way that would allow a similar outcome without need to introduce more `*_scratch_reg` variations. One option that I can think of is to remove `scratch_reg` and `float_scratch_reg` and rely exclusively on `scratch_for` and `sratch!(M, &ty)` method and macro respectively. This might make it easier to encapsulate the details of which register should be used depending on the type/size/architecture.

Reply:
Yeah I debated whether to just use the float scratch register directly but that seemed like it would be mixing different levels of abstraction. Changing everything to use `scratch_for` or `scratch!(M, &ty)` instead makes sense to me. I'll make those changes. Might try doing that in a separate PR to keep it easier to review.

Reply:
In my original comment I was leaning more towards the first option from your comment above: moving `scratch_for` to each ABI implementation, hoping to (i) get the most flexibility (architecture-wise) at the cost of some repetition (ii) reduce the number of scratch related methods from the ABI trait.

Reply:
Opened https://github.com/bytecodealliance/wasmtime/pull/8999 to do the refactoring.